### PR TITLE
NFE as a switch

### PR DESCRIPTION
**Pull request by Quick-Flash**
Adding NFE mode as a switch as well as making the logic for horizon mode better so that its harder for users to implement bad settings.

### CODE REVIEW

**Comment by nerdCopter** [comment]
@Quick-Flash , can you explain in totality how this affects Configurator (pid-tab NFE)(modes-tab), and if MSP is required for `semver` checks and value setting?

**Comment by Quick-Flash** [comment]
> @Quick-Flash , can you explain in totality how this affects Configurator (pid-tab NFE)(modes-tab), and if MSP is required for `semver` checks and value setting?

It just works in the configurator, you can flash yourself, the NFE button is now useless, and no MSP changes are needed besides removing NFE mode.

**Comment by Quick-Flash** [comment]
> Interesting, so both ANGLE and NFE need to to be enabled in configurator?

currently, would it be better if enabling NFE on its own gave the angle mode effect on roll?

File: src/main/fc/config.c [hunk 1]
```diff
@@ -236,19 +236,6 @@ static void validateAndFixConfig(void) {
...
-            pidProfilesMutable(i)->pid[PID_PITCH].F = 0;
-        }
-    }
-    if (!rcSmoothingIsEnabled() ||
-            (rxConfig()->rcInterpolationChannels != INTERPOLATION_CHANNELS_RPY &&
-             rxConfig()->rcInterpolationChannels != INTERPOLATION_CHANNELS_RPYT)) {
-        for (unsigned i = 0; i < PID_PROFILE_COUNT; i++) {
-            pidProfilesMutable(i)->pid[PID_YAW].F = 0;
-        }
-    }
```
Comment (Quick-Flash) [inline src/main/fc/config.c:251]: i forgot to mention this but i removed more of the old FF code here

**Comment by nerdCopter** [comment]
@nerdCopter reminder: to-do: if semver(firmware) <=0.3.2 {pid-tab} show NFE, else don't.

**Review by nerdCopter — APPROVED** [review]
LGTM

**Pull request merged**